Add get sheet action to google_sheets

luca-angemi opened the pull request:
<!--
  You are amazing! Thanks for contributing to our project!
  Please, DO NOT DELETE ANY TEXT from this template! (unless instructed).

  Before submitting your pull request, please verify that you have chosen the correct target branch,
  and that the PR preview looks fine and does not include unrelated changes.
-->
## Proposed change
<!-- 
    Describe the big picture of your changes here to communicate to the
    maintainers why we should accept this pull request. If it fixes a bug
    or resolves a feature request, be sure to link to that issue in the 
    additional information section.
-->



## Type of change
<!--
    What types of changes does your PR introduce to our documentation/website?
    Put an `x` in the boxes that apply. You can also fill these out after
    creating the PR.
-->

- [ ] Spelling, grammar or other readability improvements (`current` branch).
- [ ] Adjusted missing or incorrect information in the current documentation (`current` branch).
- [ ] Added documentation for a new integration I'm adding to Home Assistant (`next` branch).
  - [ ] I've opened up a PR to add logos and icons in [Brands repository](https://github.com/home-assistant/brands).
- [X] Added documentation for a new feature I'm adding to Home Assistant (`next` branch).
- [ ] Removed stale or deprecated documentation.

## Additional information
<!--
    Details are important, and help maintainers processing your PR.
    Please be sure to fill out additional details, if applicable.
-->

- Link to parent pull request in the codebase: https://github.com/home-assistant/core/pull/150133
- Link to parent pull request in the Brands repository: 
- This PR fixes or closes issue: fixes #

## Checklist
<!--
    Put an `x` in the boxes that apply. You can also fill these out after
    creating the PR. If you're unsure about any of them, don't hesitate to ask.
    We're here to help! This is simply a reminder of what we are going to look
    for before merging your code.
-->

- [X] This PR uses the correct branch, based on one of the following:
  - I made a change to the existing documentation and used the `current` branch.
  - I made a change that is related to an upcoming version of Home Assistant and used the `next` branch.
- [X] The documentation follows the Home Assistant documentation [standards].

[standards]: https://developers.home-assistant.io/docs/documenting/standards

luca-angemi committed:
Add fetch sheet action to  google_sheets

c0ffeeca7 reviewed (dismissed):
Thank you, @luca-angemi :+1:

emontnemery reviewed (dismissed):
The documentation should mention that the action returns data, for example as done here https://www.home-assistant.io/integrations/weather/#action-weatherget_forecasts
The documentation should also include an example of returned data, like this https://www.home-assistant.io/integrations/weather/#examples

emontnemery commented on source/_integrations/google_sheets.markdown:
```diff
@@ -85,3 +85,31 @@ data:
+| ---------------------- | -------- | ----------- | --------|
+| `config_entry` | no | Config entry to use. |
+| `worksheet` | yes | Name of the worksheet. Defaults to the first one in the document. | Sheet1 |
+| `rows` | no | Maximum number of rows of data to be retrieved from the worksheet.  | 2 |
```
Which rows are being returned? In the implementation PR, it seems this specifies the last n rows to return.

luca-angemi replied:
Yes, it's the last rows, up to n. I'll update.

luca-angemi committed:
Update 'rows' attribute description

Clarified description of the 'rows' data attribute.

luca-angemi committed:
Rename fetch_sheet action to get_sheet and add response example

Renamed action from `google_sheets.fetch_sheet` to `google_sheets.get_sheet` and updated related documentation.

emontnemery commented on source/_integrations/google_sheets.markdown:
```diff
@@ -85,3 +85,44 @@ data:
+| ---------------------- | -------- | ----------- | --------|
+| `config_entry` | no | Config entry to use. |
+| `worksheet` | yes | Name of the worksheet. Defaults to the first one in the document. | Sheet1 |
+| `rows` | no | Maximum number of the last rows of data to be retrieved from the worksheet.  | 2 |
```
> Maximum number of the last rows of data

Try to rephrase / reword this

emontnemery replied:
Maybe this would work:
"Maximum number of rows from the end of the worksheet to return"

luca-angemi committed:
reword

Updated description for 'rows' data attribute in Google Sheets integration documentation.

c0ffeeca7 commented on source/_integrations/google_sheets.markdown:
```diff
@@ -85,3 +85,44 @@ data:
 {% enddetails %}
+
+
+### Action `google_sheets.get_sheet`
```
```suggestion
### Action: Get sheet
```
I know the YAML version is still used across our documentation. But we are trying to move toward using a non-YAML title.
https://github.com/home-assistant/home-assistant.io/blob/current/source/_integrations/_integration_docs_template.markdown?plain=1#L139

c0ffeeca7 committed:
tiny style tweak to action title
